scripts/metrics/common/query_metrics.py

```diff
@@ -0,0 +1,157 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Provides query functionality for .csv metrics from S3 buckets
+"""
+
+import argparse
+import csv
+import os
+
+from boto3.session import Session
+
+BUCKET_NAME = "o3de-metrics"
+
+SESSION = None
+DOWNLOADED_FILES = []
+
+
+def _get_session():
+    """
+    Gets an AWS session via boto3 or returns the existing one
+    """
+    global SESSION
+    if SESSION is None:
+        # get session
+        id_ = os.environ['AWS_ACCESS_KEY_ID']
+        secret = os.environ['AWS_SECRET_ACCESS_KEY']
+        token = os.environ['AWS_SESSION_TOKEN']
+
+        SESSION = Session(
+            aws_access_key_id=id_,
+            aws_secret_access_key=secret,
+            aws_session_token=token,
+            region_name='us-west-2'
+        )
+    return SESSION
+
+
+def _download_csvs(branch, start_week, end_week, download_folder):
+    """
+    Downloads the desired csvs based on branch, start_week, and end_week to the download_folder
+    :param branch: name of the desired branch
+    :param start_week: number of the first week to include
```

Comment:
How will weeks monotonically change over time, do I need to remember some specific date and do math from that? There are 52 weeks in a year, how will I select something from two years ago? From two years after the start of data collection? From today? From a time spanning between years?

Perhaps instead of directly answering those questions, can you directly describe the levels of the folder structure somewhere?